feature-libs/pickup-in-store/components/container/my-preferred-store/my-preferred-store.component.ts

```diff
@@ -69,24 +84,78 @@ export class MyPreferredStoreComponent implements OnInit {
     );
 
     useFeatureStyles('a11yViewHoursButtonIconContrast');
+    useFeatureStyles('a11yImproveButtonsInCardComponent');
   }
 
   ngOnInit(): void {
-    this.cmsService
-      .getCurrentPage()
+    const source$ = this.cmsService.getCurrentPage().pipe(
+      filter<Page>(Boolean),
+      take(1),
+      tap(
+        (cmsPage) => (this.isStoreFinder = cmsPage.pageId === 'storefinderPage')
+      )
+    );
+
+    source$
       .pipe(
-        filter<Page>(Boolean),
-        take(1),
-        tap(
-          (cmsPage) =>
-            (this.isStoreFinder = cmsPage.pageId === 'storefinderPage')
-        ),
         filter(() => this.isStoreFinder),
         tap(() => {
-          this.content = {
+          let content: PreferredStoreContent = {
             header: '',
             actions: [{ event: 'send', name: 'Get Directions' }],
           };
+          if (
+            this.featureConfigService.isEnabled(
+              'a11yImproveButtonsInCardComponent'
+            )
+          ) {
+            const link = this.storeFinderService.getDirections(
+              this.pointOfService
+            );
+            content = {
+              ...content,
+              actions: [
+                {
+                  link,
+                  name: 'Get Directions',
+                  ariaLabel: 'cardActions.getDirections',
+                  target: '_blank',
+                },
+              ],
+            };
+          }
+          this.content = content;
+        })
+      )
+      .subscribe();
+
+    source$
```

Comment:
Do we need this second subscription? I think we can pull this off with just one

Reply:
fixed